lib/honeycomb/propagation/aws.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require "base64"
+require "json"
+require "uri"
+
+module Honeycomb
+  # Parsing and propagation for AWS trace headers
+  module AWSPropagation
+    # Parse trace headers
+    module UnmarshalTraceContext
+      def parse(serialized_trace)
+        unless serialized_trace.nil?
+          split = serialized_trace.split(";")
+
+          trace_id, parent_span_id, trace_fields = get_fields(split)
+
+          parent_span_id = trace_id if parent_span_id.nil?
+
+          trace_fields = {} if trace_fields.nil?
+
+          if !trace_id.nil? && !parent_span_id.nil?
+            return [trace_id, parent_span_id, trace_fields]
```

Comment:
I think we need an extra `nil` for dataset compared to honeycomb's implementation and the generic return below. Although, I'm not sure it's strictly necessary because ruby is _magic_.

```suggestion
            return [trace_id, parent_span_id, trace_fields, nil]
```

Reply:
good point, and I think it can't hurt. I'll add this to both w3c and aws. Even if it wouldn't necessarily break, I think handling all 4 potential fields will help readability